opengrok-indexer/src/main/java/org/opengrok/indexer/index/PendingFileCompleter.java

```diff
@@ -241,9 +266,12 @@ private int completeDeletions() throws IOException {
             new PendingFileDeletionExec(f.getAbsolutePath())).collect(
             Collectors.toList());
 
+        AtomicInteger currentCount = new AtomicInteger();
         Map<Boolean, List<PendingFileDeletionExec>> bySuccess =
             pendingExecs.parallelStream().collect(
             Collectors.groupingByConcurrent((x) -> {
+                printProgress(LOGGER, "pending deletions",
```

Comment:
It is possible due to thread intertwining that you will print higher progress before the lower progress, right? E.g. 100% will be printed and after that only 50%. Is it okay?

Reply:
not sure how big problem is this. My guess is that if anyone runs with `--progress` then he also bumps the log level higher value so the interleaving would not be as disturbing as with the extreme 100/50 case.

The solution would be to spawn a reporting thread I guess.

Reply:
Or maybe make `printProgress` `synchronized`?

Reply:
That would be more expensive that the atomic integer and could slow down the overall operation.

Reply:
I would still pick consistency over it. Very roughly, let's say lock costs `100ns`, even though you would have a thousands of operations, it still should be nothing compared to indexing time. But I'm okay with it. Should not happen that often and still, it's just a log.

Reply:
added simple logger thread.